tasks/2017/mypy-annotations.md

```diff
@@ -0,0 +1,58 @@
+# GCI Tasks: Mypy Annotations
+
+## Prerequisites
+
+* A working Zulip development environment. See
+  [here](https://github.com/zulip/zulip-gci/blob/master/README.md) for instructions
+  on how to set one up.
+
+* Update your working copy of Zulip and then create a feature branch. [Learn
+  how](../../before-every-task.md).
+
+* If this is your first contribution, you may be interested in the
+  [how to create a pull request](https://codein.withgoogle.com/tasks/6541581402243072/) and
+  [intro to Zulip server development](https://codein.withgoogle.com/tasks/4799263762546688/) tasks.
+
+## Background
+
+Mypy is an experimental static type checker for Python. Mypy ensures that
+the codebase doesn't do things like pass a string (like `'42'`) to a function
+that is expecting an int (like `42`).
+
+You can read more about how mypy works at
+http://zulip.readthedocs.io/en/latest/mypy.html.
+
+Zulip has reached 100% mypy annotations coverage as a result of
+work done during last year's code-in. Earlier, comments were used
+to specify types to support both Python 2 and 3. As we're migrating our
+codebase to Python 3 only, we can use the new PEP484 syntax for
+type annotations.
+
+Although we've built an automated script to convert most of these annotations, we
+still need to convert methods with multiline definitions manually.
+
+## Task Descriptions
+
+### Task Type A
+
+Let *file* be a file listed in the task that brought you here.
+* Convert mypy annotations for all multiline methods in the *file*.
+* Commit your changes with a commit message `mypy: User Python 3 syntax for typing in <file>.`
```

Comment:
Mention that they should ignore single-line def methods?  I'd rather not have students doing those manually, since there's more risk of bugs with manual vs. automated conversion.